DependencyInjection/Compiler/CacheCompatibilityPass.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Bundle\MongoDBBundle\DependencyInjection\Compiler;
+
+use Doctrine\Bundle\MongoDBBundle\DependencyInjection\DoctrineMongoDBExtension;
+use Doctrine\Common\Cache\CacheProvider;
+use Doctrine\Common\Cache\Psr6\CacheAdapter;
+use Doctrine\Common\Cache\Psr6\DoctrineProvider;
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\DependencyInjection\ChildDefinition;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+use function array_keys;
+use function assert;
+use function is_a;
+use function is_string;
+use function trigger_deprecation;
+
+/** @internal */
+final class CacheCompatibilityPass implements CompilerPassInterface
+{
+    private const CACHE_SETTER_METHODS_PSR6_SUPPORT = [
+        'setMetadataCache' => true,
+        'setMetadataCacheImpl' => true,
+    ];
+
+    public function process(ContainerBuilder $container): void
+    {
+        foreach (array_keys($container->findTaggedServiceIds(DoctrineMongoDBExtension::CONFIGURATION_TAG)) as $id) {
+            /** @var array<int, string|Reference[]> $methodCall */
+            foreach ($container->getDefinition($id)->getMethodCalls() as $methodCall) {
+                $methodName = $methodCall[0];
+                assert(is_string($methodName));
+
+                if (! isset(self::CACHE_SETTER_METHODS_PSR6_SUPPORT[$methodName])) {
+                    continue;
+                }
+
+                /** @var Reference[] $methodArgs */
+                $methodArgs   = $methodCall[1];
+                $definitionId = (string) $methodArgs[0];
+                $aliasId      = null;
+                if ($container->hasAlias($definitionId)) {
```

Comment:
So this is where I deviate the most from the doctrinebundle, because as doctrinebundle expects to get an alias here, I actually didn't find an alias for the method argument. <br>
Figured it wasn't actually relevant for the whole wrapping anyway, so just go with whatever's given and make sure to update the alias, if the argument represents an alias.

Reply:
Ping @ostrolucky as he was the one that came up with the logic after I failed to do so. Maybe he can provide some insight into this.

Reply:
Realised that the definitionId should then be the aliasId